eth/handler_wit.go

```diff
@@ -182,6 +197,96 @@ func (h *witHandler) handleGetWitness(peer *wit.Peer, req *wit.GetWitnessPacket)
 	return response, nil
 }
 
+// handleGetReducedWitness retrieves witnesses and filters them using the sliding window cache.
+// This reduces bandwidth by omitting state nodes that the receiver should already have cached.
+func (h *witHandler) handleGetReducedWitness(peer *wit.Peer, req *wit.GetWitnessPacket) (wit.WitnessPacketResponse, error) {
```

Comment:
The result could be cached to reduce computation.

Reply:
Does [this](https://github.com/0xPolygon/bor/pull/1818/commits/bee613647511c1778d25960cf4361e1bab510f50) look good? Unsure about the size of the cache though, it will be ok to cache blocks > length of the window, right?